core/dbt/cli/flags.py

```diff
@@ -349,6 +353,24 @@ def _assert_mutually_exclusive(
             elif flag_set_by_user:
                 set_flag = flag
 
+    def _validate_event_time_configs(self) -> None:
+        event_time_start = (
+            getattr(self, "EVENT_TIME_START") if hasattr(self, "EVENT_TIME_START") else None
+        )
+        if event_time_start is not None:
```

Comment:
Nit: do we need this level of nesting?

Reply:
I think so? We need to check whether or not `event_time_start` is `None`, as it only makes sense to do the validation if `event_time_start` is a datetime value. We could get around this layer of nesting by doing something like:
```python
if not hasattr(self, "EVENT_TIME_START"):
  return

event_time_start = getattr(self, "EVENT_TIME_START")
...
```

I'm personally not a huge fan of multiple returns. I can see how doing an early return would reduce our level of if's from 2 down to 1. However, given we're only at 2 levels of if's that seems unnecessary.

---
Side note that's not applicable here, but a corollary to my above "I avoid multiple returns".

If there were more levels of nesting, I still likely woudn't add early returns. I'd instead end up looking at splitting it into multiple functions, thus continuing allowing to maintain low levels of if context nesting and avoiding early multiple returns in a function.